src/main/java/com/endava/soa_5/tools/PropertyReader.java

```diff
@@ -0,0 +1,35 @@
+package com.endava.soa_5.tools;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+import static com.endava.soa_5.tools.Constants.CONFIGURATION_FOLDER;
+
+public class PropertyReader {
+    private Properties properties;
+    private InputStream inputStream;
+    private String propertyValue;
+
+    public PropertyReader() {
+        properties = new Properties();
+    }
+
+    public String getPropertyValue(String key) {
+        try {
+            inputStream = getClass().getClassLoader().getResourceAsStream(CONFIGURATION_FOLDER + "Configuration.properties");
+            properties.load(inputStream);
+
```

Comment:
you can optimize your code by splitting the action of reading file and getting the property into two separate methods,
then calling the reading method into the constructor(readProperties() for example).
That way, when running the program, the property file will only be read once and stored into an object, which we can use to extract values from (using the getProperty method)

Reply:
Implemented in another way.